packages/std/src/math/decimal.rs

```diff
@@ -184,6 +184,19 @@ impl Decimal {
         Self::DECIMAL_PLACES as u32
     }
 
+    pub fn floor(&self) -> Self {
+        Self((self.0 / Self::DECIMAL_FRACTIONAL) * Self::DECIMAL_FRACTIONAL)
+    }
+
+    pub fn ceil(&self) -> Self {
```

Comment:
In contrast to `floor`, this can fail. For all decimals > largest possible integer the result cannot be stored anymore. You can test `Decimal::MAX.ceil()`.

Not sure how to best handle that case. We can panic or return an `Option`. But we should test and document it.

Reply:
I was actually thinking about automatic panic on addition. Good point on test, I'll add doc comment as well.

Reply:
What about having a panicking `ceil` and a `checked_ceil`?

Reply:
Implemented.